packages/internal-test-env/index.ts

```diff
@@ -143,113 +141,147 @@ function getTestingEnvironment(
   }
 }
 
+function getBaseTestingEnvironment<T extends LibraryVariables>(libVars?: T
+  ): T extends NodeVariables ? TestingEnvironmentNode : TestingEnvironmentBrowser   {
+    setupEnv();
+    getTestingEnvironment(process.env);
+  
+    const base = {
+      idp: process.env.E2E_TEST_IDP,
+      environment: process.env.E2E_TEST_ENVIRONMENT,
+      features: featuredFlags,
+    };
+  
+    return libVars ? merge(base, validateLibVars(libVars)) : base;
+}
+
 export function getNodeTestingEnvironment(
   libVars?: LibraryVariables
 ): TestingEnvironmentNode {
-  setupEnv();
-  getTestingEnvironment(process.env);
-
-  if (
-    process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== undefined ||
-    process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== undefined
-  ) {
-    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== "string") {
-      throw new Error(
-        "The environment variable E2E_TEST_REQUESTOR_CLIENT_ID is undefined."
-      );
-    }
-
-    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
-      throw new Error(
-        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
-      );
-    }
-  }
-
-  const base = {
-    idp: process.env.E2E_TEST_IDP,
-    environment: process.env.E2E_TEST_ENVIRONMENT,
-    clientCredentials: {
-      owner: {
-        id: process.env.E2E_TEST_OWNER_CLIENT_ID,
-        secret: process.env.E2E_TEST_OWNER_CLIENT_SECRET,
-      },
-    },
-    features: featuredFlags,
-  };
+  return getBaseTestingEnvironment({
+    ...libVars,
+    // Enforce client credentials are present for the resource owner.
+    clientCredentials: { owner: { id: true, secret: true }}
+  });
```

Comment:
This needs to be a merge deep, as we're dealing with nested objects.

```suggestion
  return getBaseTestingEnvironment(
     merge(desiredVars, {
       clientCredentials: { owner: { id: true, secret: true }}
     })
  );
```

As otherwise requesting the browser testing environment, but also wanting the client credentials id/secret pair will fail / requesting the `requestor` credentials will also fail, as `clientCredentials` will be overwritten from libVars with the contents of `clientCredentials` on line 164.

Reply:
Oh nice catch, that would have been a pain to debug. Fixed in 4117ab386905fcb35518fc7465eb266458882eb6